From 2710b6453f3ac88badb4c529400b4af1b4cefaf3 Mon Sep 17 00:00:00 2001
From: Carl Kittelberger <icedream@icedream.pw>
Date: Tue, 26 Sep 2023 16:46:11 +0200
Subject: [PATCH] Apply patches.

---
 PKGBUILD | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/PKGBUILD b/PKGBUILD
index 6c0bef9..987c21a 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -26,6 +26,13 @@ pkgver() {
   echo "${_ver}_r$(git rev-list --count HEAD).g$(git rev-parse --short HEAD)"
 }
 
+prepare() {
+  cd $pkgname-$pkgver
+  for patch in "${srcdir}"/*-*.patch; do
+    patch -p1 -i "$patch"
+  done
+}
+
 build() {
   cmake -B build -S ${pkgname%-git} \
     -DQT_MAJOR_VERSION=6 \
-- 
2.42.0

