From c955791491d551c644f14e3c23de1c526f01be5f Mon Sep 17 00:00:00 2001
From: Carl Kittelberger <icedream@icedream.pw>
Date: Wed, 8 Jun 2022 20:46:41 +0200
Subject: [PATCH] Add missing dependency for pyside2.

---
 PKGBUILD | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/PKGBUILD b/PKGBUILD
index 56e1cac..c7e5adb 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -7,7 +7,7 @@ pkgdesc="A tool to let users always have up-to-date public keys for other member
 arch=('any')
 url="https://github.com/firstlookmedia/gpgsync/"
 license=('GPL3')
-depends=('python-setuptools' 'python-pyqt5' 'python-nose' 'python-requests' 'python-pysocks' 'python-packaging' 'python-dateutil' 'gnupg')
+depends=('python-setuptools' 'python-pyqt5' 'python-nose' 'python-requests' 'python-pysocks' 'python-packaging' 'python-dateutil' 'pyside2' 'gnupg')
 source=("$pkgname-$pkgver.tar.gz::https://github.com/firstlookmedia/${pkgname}/archive/v${pkgver}.tar.gz" gpgsync.{service,timer})
 sha512sums=('341dd08853929b50adff5247decb8844dafa0dbe43ccb0541f6578815c78a6019a27daa15d913e43200f2dca4567e215658b357408f1866536729e78c95615b8'
             '26a56775912d680ce34c32c5e24a3dd5826b3ea78c6c3d4295d8da199d881b30c200631fd0a49077c3b4617e4941cbcd57a342e535939ea04fe4e489f15f9d18'
-- 
2.36.1

