From d648f4d2cc0579fafee80e6489ce1c58877f1e3c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Daniel=20Sch=C3=BCrmann?= <daschuer@mixxx.org>
Date: Thu, 8 Feb 2024 13:54:47 +0100
Subject: [PATCH 4/5] Use taglib1path on Linux only

---
 cmake/modules/FindTagLib.cmake | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/cmake/modules/FindTagLib.cmake b/cmake/modules/FindTagLib.cmake
index 1179ebb7f2..923143715e 100644
--- a/cmake/modules/FindTagLib.cmake
+++ b/cmake/modules/FindTagLib.cmake
@@ -47,8 +47,10 @@ include(IsStaticLibrary)
 
 find_package(PkgConfig QUIET)
 if(PkgConfig_FOUND)
-  # priorize the taglib1 package introduced in https://aur.archlinux.org/packages/taglib1
-  set(ENV{PKG_CONFIG_PATH} "/usr/lib/taglib1/lib/pkgconfig/")
+  if(UNIX AND NOT APPLE)
+    # priorize the taglib1 package introduced in https://aur.archlinux.org/packages/taglib1
+    set(ENV{PKG_CONFIG_PATH} "/usr/lib/taglib1/lib/pkgconfig/:$ENV{PKG_CONFIG_PATH}")
+  endif()
   pkg_check_modules(PC_TagLib QUIET taglib)
 endif()
 
-- 
2.43.0

