From e7c99be14cf21661cc5861c615630ebc0988dbab Mon Sep 17 00:00:00 2001
From: Carl Kittelberger <icedream@icedream.pw>
Date: Sat, 10 Feb 2024 22:52:39 +0100
Subject: [PATCH 5/5] Add TagLib as PUBLIC

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f59cb3ab0f..99f0e47c43 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2781,7 +2781,7 @@ target_link_libraries(mixxx-lib PRIVATE SoundTouch::SoundTouch)
 
 # TagLib
 find_package(TagLib 1.11 REQUIRED)
-target_link_libraries(mixxx-lib PRIVATE TagLib::TagLib)
+target_link_libraries(mixxx-lib PUBLIC TagLib::TagLib)
 
 # Threads
 set(THREADS_PREFER_PTHREAD_FLAG ON)
-- 
2.43.0

