diff --git a/PKGBUILD b/PKGBUILD
index d54241b..7e26f67 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -11,7 +11,7 @@ license=('GPL')
 groups=('kodi')
 makedepends=('cmake' 'git' 'pkgconf' 'kodi-dev')
 provides=('kodi-addon-inputstream-ffmpegdirect')
-depends=('kodi')
+depends=('kodi' 'ffmpeg5.1')
 source=("https://github.com/xbmc/${_gitname}/archive/${pkgver}-${_koditarget}.tar.gz")
 
 sha256sums=('b28ccbd489d9b69779fd818ee158f69ebd6fd85b1f0410ffc8098ff0c8665bdc')
@@ -23,6 +23,7 @@ build() {
     cmake \
         -DCMAKE_INSTALL_PREFIX=/usr \
         -DCMAKE_BUILD_TYPE=Release \
+        -DFFMPEG_INCLUDE_DIRS=/usr/include/ffmpeg5.1 \
         ..
     make
 }
