From 0c10647438252f0920be915a64e1d5b18888e033 Mon Sep 17 00:00:00 2001
From: Carl Kittelberger <icedream@icedream.pw>
Date: Wed, 22 Feb 2023 01:52:36 +0100
Subject: [PATCH] Fix missing Vulkan extensions.

---
 PKGBUILD | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/PKGBUILD b/PKGBUILD
index fb44c0d..456a2f3 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -8,8 +8,8 @@
 
 _name=ffmpeg
 pkgname=ffmpeg-libfdk_aac
-pkgver=5.1.2
-pkgrel=2
+pkgver=5.1.2.r1.abb591bcc9
+pkgrel=1
 epoch=1
 pkgdesc='Complete solution to record, convert and stream audio and video (Same as official package except with libfdk-aac support)'
 arch=(x86_64)
@@ -123,11 +123,16 @@ validpgpkeys=(DD1EC9E8DE085C629B3E1846B18E8928B3948D64) # Michael Niedermayer <m
 prepare() {
   cd ffmpeg
   patch -Np1 -i ../add-av_stream_get_first_dts-for-chromium.patch # https://crbug.com/1251779
+
+  if [ "x$( git branch --contains eb0455d646 $(git branch --show-current) )" = "x" ]; then
+    # apply fix for Vulkan extensions
+    git cherry-pick eb0455d646 # solve missing Vulkan extensions
+  fi
 }
 
 pkgver() {
   cd ffmpeg
-  git describe --tags | sed 's/^n//'
+  git describe --tags | sed -e 's/^n//' -e 's/\([^-]*-\)g/r\1/;s/-/./g'
 }
 
 build() {
-- 
2.39.2

