From 168aa86a98954925334fdbab9b3cbbb8447b8093 Mon Sep 17 00:00:00 2001
From: Carl Kittelberger <icedream@icedream.pw>
Date: Thu, 23 May 2024 23:25:14 +0200
Subject: [PATCH 2/4] Add libftdi and libusb as dependencies.

---
 PKGBUILD | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/PKGBUILD b/PKGBUILD
index af142ca..c8b5985 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -7,8 +7,8 @@ pkgdesc="ectool for the Framework laptop."
 arch=(x86_64)
 url="https://www.howett.net/posts/2021-12-framework-ec/#software"
 provides=('ectool')
-depends=()
-makedepends=('inetutils' 'git')
+depends=('libftdi' 'libusb')
+makedepends=('inetutils' 'git' 'cmake')
 license=('BSD')
 source=(git+https://gitlab.howett.net/DHowett/ectool.git)
 sha1sums=('SKIP')
-- 
2.45.1

