From 0bee0acd61df70ecbd229ab95e368ac3b344c08d Mon Sep 17 00:00:00 2001
From: Carl Kittelberger <icedream@icedream.pw>
Date: Thu, 23 May 2024 23:25:21 +0200
Subject: [PATCH 3/4] Update pkgver.

---
 PKGBUILD | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/PKGBUILD b/PKGBUILD
index c8b5985..da60739 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -1,7 +1,7 @@
 # Maintainer: Cole Deck <cole at deck dot sh>
 pkgname=fw-ectool-git
 _gitname=ectool
-pkgver=r2760.3ebe7b8
+pkgver=r2762.39d64fb
 pkgrel=1
 pkgdesc="ectool for the Framework laptop."
 arch=(x86_64)
-- 
2.45.1

