From 0eb30b8330acfe0a7fa3e53dadfd274e44f853e5 Mon Sep 17 00:00:00 2001
From: c-basalt <117849907+c-basalt@users.noreply.github.com>
Date: Fri, 9 Aug 2024 18:23:25 -0400
Subject: [PATCH 03/14] fix

---
 yt_dlp/extractor/rplaylive.py | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/yt_dlp/extractor/rplaylive.py b/yt_dlp/extractor/rplaylive.py
index cec0ae670..b90e9ef80 100644
--- a/yt_dlp/extractor/rplaylive.py
+++ b/yt_dlp/extractor/rplaylive.py
@@ -139,8 +139,6 @@ class RPlayVideoIE(RPlayBaseIE):
     def _real_extract(self, url):
         video_id = self._match_id(url)
 
-        self.get_butter_token()
-
         playlist_id = traverse_obj(parse_qs(url), ('playlist', ..., any))
         if playlist_id and self._yes_playlist(playlist_id, video_id):
             playlist_info = self._download_json(
@@ -170,7 +168,7 @@ def _real_extract(self, url):
             'release_timestamp': ('publishedAt', {parse_iso8601}),
             'timestamp': ('createdAt', {parse_iso8601}),
             'duration': ('length', {float_or_none}),
-            'uploader': ('nickname', {str}),
+            'uploader': ('creatorInfo', 'nickname', {str}),
             'uploader_id': ('creatorOid', {str}),
             'tags': ('hashtags', lambda _, v: v[0] != '_'),
             'age_limit': (('hideContent', 'isAdultContent'), {lambda x: 18 if x else None}, any),
-- 
2.52.0

