From 1b0debd5838e56a82294d7406b791a06d4517937 Mon Sep 17 00:00:00 2001
From: c-basalt <117849907+c-basalt@users.noreply.github.com>
Date: Sun, 29 Dec 2024 03:24:11 -0500
Subject: [PATCH 10/14] hardcode logintype

---
 yt_dlp/extractor/rplaylive.py | 9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/yt_dlp/extractor/rplaylive.py b/yt_dlp/extractor/rplaylive.py
index c75f8a83c..a574fc42c 100644
--- a/yt_dlp/extractor/rplaylive.py
+++ b/yt_dlp/extractor/rplaylive.py
@@ -25,7 +25,6 @@ class RPlayBaseIE(InfoExtractor):
     _NETRC_MACHINE = 'rplaylive'
     _TOKEN_CACHE = {}
     _user_id = None
-    _login_type = None
     _jwt_token = None
     _tested_jwt = False
 
@@ -40,11 +39,6 @@ def user_id(self):
         self._check_jwt_args()
         return self._user_id
 
-    @property
-    def login_type(self):
-        self._check_jwt_args()
-        return self._login_type
-
     @property
     def jwt_token(self):
         self._check_jwt_args()
@@ -54,7 +48,7 @@ def jwt_token(self):
     def requestor_query(self):
         return {
             'requestorOid': self.user_id,
-            'loginType': self.login_type,
+            'loginType': 'plax',
         } if self.user_id else {}
 
     @property
@@ -106,7 +100,6 @@ def _login_by_token(self, jwt_token, raw_token_hint=False):
 
         if user_info:
             self._user_id = traverse_obj(user_info, 'oid')
-            self._login_type = traverse_obj(user_info, 'accountType')
             self._jwt_token = jwt_token if self._user_id else None
         if not self._user_id:
             if raw_token_hint:
-- 
2.52.0

