From 23cc6e564d0485682fba63320a7141145f73c5ce Mon Sep 17 00:00:00 2001
From: c-basalt <117849907+c-basalt@users.noreply.github.com>
Date: Tue, 3 Mar 2026 11:09:03 -0500
Subject: [PATCH 18/18] update api endpoint

---
 yt_dlp/extractor/rplaylive.py | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/yt_dlp/extractor/rplaylive.py b/yt_dlp/extractor/rplaylive.py
index eb79d41ed..967ca3ecb 100644
--- a/yt_dlp/extractor/rplaylive.py
+++ b/yt_dlp/extractor/rplaylive.py
@@ -100,7 +100,7 @@ def _perform_login(self, username, password):
 
     def _login_by_token(self, jwt_token, raw_token_hint=False):
         user_info = self._download_json(
-            'https://api.rplay-cdn.com/account/login', 'login', note='performing login', errnote='login failed',
+            'https://api.rplay.live/account/login', 'login', note='performing login', errnote='login failed',
             data=f'{{"token":"{jwt_token}","loginType":null,"checkAdmin":null}}'.encode(),
             headers={'Content-Type': 'application/json', 'Authorization': 'null'}, fatal=False)
 
@@ -216,7 +216,7 @@ def _real_extract(self, url):
         playlist_id = traverse_obj(parse_qs(url), ('playlist', ..., any))
         if playlist_id and self._yes_playlist(playlist_id, video_id):
             playlist_info = self._download_json(
-                'https://api.rplay-cdn.com/content/playlist', playlist_id,
+                'https://api.rplay.live/content/playlist', playlist_id,
                 query={'playlistOid': playlist_id, **self.requestor_query},
                 headers=self.jwt_header, fatal=False)
             if playlist_info:
@@ -226,7 +226,7 @@ def _real_extract(self, url):
             else:
                 self.report_warning('Failed to get playlist, downloading video only')
 
-        video_info = self._download_json('https://api.rplay-cdn.com/content', video_id, query={
+        video_info = self._download_json('https://api.rplay.live/content', video_id, query={
             'contentOid': video_id,
             'status': 'published',
             'withComments': True,
@@ -321,10 +321,10 @@ class RPlayUserIE(InfoExtractor):
     def _real_extract(self, url):
         user_id, short = self._match_valid_url(url).group('id', 'short')
 
-        user_info = self._download_json('https://api.rplay-cdn.com/account/getuser', user_id, query={
+        user_info = self._download_json('https://api.rplay.live/account/getuser', user_id, query={
             'customUrl' if short == 'c' else 'userOid': user_id, 'options': '{"includeContentMetadata":true}'})
         replays = self._download_json(
-            'https://api.rplay-cdn.com/live/replays', user_id, query={'creatorOid': user_info.get('_id')})
+            'https://api.rplay.live/live/replays', user_id, query={'creatorOid': user_info.get('_id')})
 
         def _entries():
             def _entry_ids():
@@ -367,11 +367,11 @@ class RPlayLiveIE(RPlayBaseIE):
     def _real_extract(self, url):
         user_id, short = self._match_valid_url(url).group('id', 'short')
 
-        user_info = self._download_json('https://api.rplay-cdn.com/account/getuser', user_id, query={
+        user_info = self._download_json('https://api.rplay.live/account/getuser', user_id, query={
             'customUrl' if short == 'c' else 'userOid': user_id})
         user_id = user_info['_id']
 
-        live_info = self._download_json('https://api.rplay-cdn.com/live/play', user_id, query={
+        live_info = self._download_json('https://api.rplay.live/live/play', user_id, query={
             'creatorOid': user_id, **self.requestor_query}, headers=self.jwt_header)
 
         stream_state = live_info['streamState']
@@ -389,12 +389,12 @@ def _real_extract(self, url):
                     raise ExtractorError('The livestream requires a free subscription to access', expected=True)
                 raise ExtractorError('You do not have access to the livestream', expected=True)
             key2 = traverse_obj(self._download_json(
-                'https://api.rplay-cdn.com/live/key2', user_id, 'getting live key',
+                'https://api.rplay.live/live/key2', user_id, 'getting live key',
                 headers=self.jwt_header, query=self.requestor_query), ('authKey', {str})) if self.user_id else ''
             if key2 is None:
                 raise ExtractorError('Failed to get playlist key')
             formats = self._extract_m3u8_formats(
-                'https://api.rplay-cdn.com/live/stream/playlist.m3u8', user_id,
+                'https://api.rplay.live/live/stream/playlist.m3u8', user_id,
                 query={'creatorOid': user_id, 'key2': key2}, headers={'Referer': 'https://rplay.live'})
 
             return {
-- 
2.53.0

