From 46bfbd0bd6937914a9f3273077d7753e8b50c909 Mon Sep 17 00:00:00 2001
From: Mozi <29089388+pzhlkj6612@users.noreply.github.com>
Date: Fri, 31 Jan 2025 11:35:05 +0000
Subject: [PATCH 09/12] wip

---
 yt_dlp/extractor/sheeta.py | 23 ++++++++++++-----------
 1 file changed, 12 insertions(+), 11 deletions(-)

diff --git a/yt_dlp/extractor/sheeta.py b/yt_dlp/extractor/sheeta.py
index 868623e50..8221d5e1b 100644
--- a/yt_dlp/extractor/sheeta.py
+++ b/yt_dlp/extractor/sheeta.py
@@ -95,7 +95,7 @@ def _refresh_token(self):
             })
 
         res = self._ie._download_json(
-            **refresh_func_params, expected_status=(400, 403, 404),
+            **refresh_func_params, video_id=None, expected_status=(400, 403, 404),
             note='Refreshing token', errnote='Unable to refresh token')
         if error := traverse_obj(
                 res, ('error', 'message', {lambda x: base64.b64decode(x).decode()}), ('error', 'message')):
@@ -167,7 +167,6 @@ def _niconico_sns_login(self, redirect_url, refresh_url):
         self._auth_info = {
             'refresh_func_params': {
                 'url_or_request': refresh_url,
-                'video_id': None,
                 'headers': {'Authorization': auth_token},
                 'data': b'',
             },
@@ -271,15 +270,17 @@ def _auth0_login(self):
             'version': '2.0.6',
         }).encode()).decode()
 
-        self._auth_info = {'refresh_func_params': {
-            'url_or_request': token_url,
-            'video_id': None,
-            'headers': {'Auth0-Client': auth0_client},
-            'data': {
-                'client_id': auth0_web_client_id,
-                'grant_type': 'refresh_token',
-                'redirect_uri': redirect_url,
-            }}}
+        self._auth_info = {
+            'refresh_func_params': {
+                'url_or_request': token_url,
+                'headers': {'Auth0-Client': auth0_client},
+                'data': {
+                    'client_id': auth0_web_client_id,
+                    'grant_type': 'refresh_token',
+                    'redirect_uri': redirect_url,
+                },
+            },
+        }
 
         def random_str():
             return ''.join(random.choices(string.digits + string.ascii_letters, k=43))
-- 
2.53.0

