From ad84702435a206d2ff6e96155d68ab1693f1d86e Mon Sep 17 00:00:00 2001
From: Carl Kittelberger <icedream@icedream.pw>
Date: Fri, 15 Dec 2023 03:05:24 +0100
Subject: [PATCH] replace vcn 4.0.2 firmware with testing firmware for vp9
 decoding issue

---
 PKGBUILD | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/PKGBUILD b/PKGBUILD
index e1f6316..ef4c0b5 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -14,8 +14,10 @@ license=('GPL2' 'GPL3' 'custom')
 arch=('any')
 makedepends=('git' 'rdfind')
 options=(!strip)
-source=("git+https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git#tag=${_tag}?signed")
-sha256sums=('SKIP')
+source=("git+https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git#tag=${_tag}?signed"
+        "https://gitlab.freedesktop.org/mesa/mesa/uploads/f51d221a24d4ac354e2d1d901613b594/vcn_4_0_2.bin")
+sha256sums=('SKIP'
+            'e88f38f4bc271a0e98be2372f88613ef9d563076af5b66cad0c6eea84dd5e465')
 validpgpkeys=('4CDE8575E547BF835FE15807A31B6BD72486CFD6') # Josh Boyer <jwboyer@fedoraproject.org>
 
 _backports=(
@@ -29,6 +31,12 @@ prepare() {
     git log --oneline -1 "${_c}"
     git cherry-pick -n "${_c}"
   done
+
+
+  # HACK - replace AMD VCN 4.0.2 with testing firmware for fixing VP9 decoding issues
+  if [ "$(sha256sum ./amdgpu/vcn_4_0_2.bin | awk '{print $1}')" = "55cbfb7ef700d2ee006a6a47cfc87d24706c7334caba6a8ce3460a1e76e26578" ]; then
+    cp "${srcdir}/vcn_4_0_2.bin" ./amdgpu
+  fi
 }
 
 pkgver() {
-- 
2.43.0

