From c45eecfb37deb7971c30ef8c6729501656cf8ee7 Mon Sep 17 00:00:00 2001
From: Carl Kittelberger <icedream@icedream.pw>
Date: Mon, 31 Jan 2022 22:01:16 +0100
Subject: [PATCH] Enable libfdk-aac.

---
 PKGBUILD | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/PKGBUILD b/PKGBUILD
index 301279e..106825c 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -23,6 +23,7 @@ depends=(
   libbluray.so
   libdav1d.so
   libdrm
+  libfdk-aac.so
   libfreetype.so
   libiec61883
   libmfx
@@ -154,6 +155,7 @@ build() {
     --enable-libbluray \
     --enable-libdav1d \
     --enable-libdrm \
+    --enable-libfdk-aac \
     --enable-libfreetype \
     --enable-libfribidi \
     --enable-libgsm \
-- 
2.35.1

