From f2c0201df3b80a9e41dd99ed21a7623a9f146452 Mon Sep 17 00:00:00 2001
From: Carl Kittelberger <icedream@icedream.pw>
Date: Mon, 14 Aug 2023 11:22:22 +0200
Subject: [PATCH 1/2] Fix pkg vars not detected if on first line.

---
 customizepkg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/customizepkg b/customizepkg
index efa5bf4..558ee0a 100755
--- a/customizepkg
+++ b/customizepkg
@@ -165,7 +165,7 @@ fi
 # use eval instead of creating a temp file to get pkgname etc
 # second grep used to filter out all variables with subshell execution like $() or ``
 # and pkgdesc variable, because multiline description breaks eval execution
-eval $(grep -Pazo '[^[:print:]][[:blank:]]*_?(pkg.*|name)=(\((.|\n)*?\)|[^#]*?(?= *#|\x0a))' ./PKGBUILD | grep -Eva '\$\(|`|pkgdesc' | tr -d '\0')
+eval $(grep -Pazo '([^[:print:]]|^)[[:blank:]]*_?(pkg.*|name)=(\((.|\n)*?\)|[^#]*?(?= *#|\x0a))' ./PKGBUILD | grep -Eva '\$\(|`|pkgdesc' | tr -d '\0')
 
 # copy for modification
 cp ./PKGBUILD ./PKGBUILD.custom
-- 
2.43.1

