From 1b3d9044c9635b85361a67bb381dd8050e50dac6 Mon Sep 17 00:00:00 2001
From: Carl Kittelberger <icedream@icedream.pw>
Date: Wed, 29 Nov 2023 04:44:01 +0100
Subject: [PATCH 2/2] Use pkgbase variable.

---
 customizepkg | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/customizepkg b/customizepkg
index 951e676..a01f1bb 100755
--- a/customizepkg
+++ b/customizepkg
@@ -33,7 +33,8 @@ usage() {
 			"${NAME} --vimdiff, -v  show diff between customised file and\n" \
 			"                            original file with vimdiff\n" \
 			"\n" \
-			"configuration is read from:\n $LOCALCONFIGDIR/\${pkgname} or $CONFIGDIR/\${pkgname}\n" \
+			"configuration is read from:\n $LOCALCONFIGDIR/\${pkgbase:-pkgname} " \
+                        "or $CONFIGDIR/\${pkgbase:-pkgname}\n" \
 			"(see $CONFIGDIR/mplayer.example as an example)\n" \
 			"\n" \
 			"originally written by <wain@archlinux.fr>\n" \
@@ -170,6 +171,20 @@ eval $(grep -Pazo '([^[:print:]]|^)[[:blank:]]*_?(pkg.*|name)=(\((.|\n)*?\)|[^#]
 # copy for modification
 cp ./PKGBUILD ./PKGBUILD.custom
 
+if [ -n "${pkgbase:-}" ]; then
+	already_included=0
+	for foundpkgname in "${pkgname[@]}"; do
+		if [ "$foundpkgname" != "$pkgbase" ]; then
+			continue
+		fi
+		already_included=1
+		break
+	done
+	if [ "$already_included" -eq 0 ]; then
+		pkgname=("$pkgbase" "${pkgname[@]}")
+	fi
+fi
+
 for package in "${pkgname[@]}"
 do
 	# local user prefs take priority
-- 
2.43.0

