From c5738dc9fd0e9169136eebf31686577153bb3b96 Mon Sep 17 00:00:00 2001
From: Carl Kittelberger <icedream@icedream.pw>
Date: Mon, 6 Sep 2021 12:36:02 +0200
Subject: [PATCH] Use system-installed libndi includes.

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5c2ff5e..d982378 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -49,7 +49,7 @@ add_library(obs-ndi MODULE
 	${obs-ndi_HEADERS})
 
 include_directories(
-	"lib/ndi"
+	"/usr/include"
 	"${LIBOBS_INCLUDE_DIR}/../UI/obs-frontend-api"
 	${Qt5Core_INCLUDES}
 	${Qt5Widgets_INCLUDES})
-- 
2.33.0

